### ISSUE TITLE
How to clip weights of a dense layer between every step of training op?

### ISSUE THREAD

**Issue by jingwb222** [tensorflow/tensorflow#21679]
Is there any possible way to do a custom op (for example, to clip by values of the weights of a dense layer manually every training step)

    if mode == tf.estimator.ModeKeys.TRAIN:
        optimizer = tf.train.AdamOptimizer(learning_rate=0.001)
        weights = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES, scope='logits/kernel')[0]
        clip_op = tf.assign(weights, tf.clip_by_value(weights, 0.01, 0.1))
        train_op = optimizer.minimize(
            loss=loss,
            global_step=tf.train.get_global_step())
        return tf.estimator.EstimatorSpec(mode=mode, loss=loss, train_op=train_op)

Above, I want to clip the weights of `logits/kernel` layer between every training op, but it does not work as intended, the weights of that layer would still drift outside the range of (0.01, 0.1). I wonder what I'm missing here.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jingwb222** [comment]
Sorry for the delayed response. I somehow managed to get a walkaround solution with `kernel_constraint` arguments. This issue can be closed for now I guess.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatatodd: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by harshini-gadige** [comment]
Closing this issue as said by @jingwb222
